### DISCUSSION TITLE
Why PVC is also getting created when using copyMethod Snapshot?

### DISCUSSION THREAD

**Discussion by winkrs** [backube/volsync#1113] [Q&A] [answered]
From what I understand, when a copyMethod `Snapshot` is being used and the `destinationPVC` is not specified then a snapshot of the source PVC will be created. Then, we can create a new PVC and uses the volume populator to reference the volume snapshot as the data source. However, what I do not understand is that along with the volumeSnapshot, a PVC is also created in the destination. Can anyone explain why is it this way?

**Comment by tesshuflower** [comment, marked as answer]
@winkrs The destination PVC still needs to be created as this is where the data is copied to during the sync operation on the `replicationdestination`.  Note that if we're talking about the `replicationdestination`, then the snapshot is actually a copy of the destination PVC after the sync operation has completed. 

The source PVC would only be snapshotted at the `replicationsource` side.

Steps on the destination:
- check if destinationPVC exists, or if not specified, create a destination PVC to write data to
- perform the sync operation - the destination mover pod will write to the destination PVC
- Perform a snapshot of this destination PVC

At this point if you had a replication destination that is doing syncs on a schedule, it's desirable to keep the destination PVC around, as it will already have a copy of the data, and then only the delta can be sent (a good example for where this is useful is rsync).

If you delete a replicationdestination then volsync will clean up the temporary destination pvc if it had originally created it.  If you're using the volumepopulator and have provisioned a pvc pointing to that replicationdestination - once that pvc is provisioned you are free to delete the replicationdestination associated with it.

**Reply by winkrs** [reply]
okay, its making sense now. Thank you for clearing my doubt.
